consensus/docker-entrypoint.sh

```diff
@@ -4,8 +4,12 @@ set -e
 echo "consensus RUN_TYPE: $RUN_TYPE"
 
 if [ "$RUN_TYPE" = "SENDER" ] ; then
-  echo "Starting in SENDER mode"
-  exec python3 -u sender.py
+  if [ "$BN_CONSENSUS_PRIVATE_KEY" != "" ] ; then
+    echo "Starting in SENDER mode"
+    exec python3 -u sender.py
+  else
+    echo "Ignore starting sender because BN_CONSENSUS_PRIVATE_KEY is not set"
+  fi
```

Comment:
Why does it matter whether or not a read-only node starts sender?  Are we try to saving resources from not running this service or are we trying to save operator time by not requiring generating a key pair they're not going to use?

Reply:
In fact @TripleSpeeder introduced the `read-only node` term in Dappnode package README file.
I think the main reason was to save operator time by not requiring generating a key pair they're not going to use and sending Eidi to that.
I just wanted added this to ensure exceptions will not be raised from `consensus_sender` if `BN_CONSENSUS_PRIVATE_KEY` var is not which is current dapp node configuration.